src/tool/cppwinrt/cppwinrt/code_writers.h

```diff
@@ -373,20 +358,14 @@ namespace xlang
 
         if (empty(generics))
         {
-            auto format = R"(    template <> struct name<%>
-    {
-        static constexpr auto & value{ L"%.%" };
-    };
+            auto format = R"(    template <> constexpr auto& name_v<%>{ L"%.%" };
```

Comment:
Should these constexpr variable templates be declared `inline`?

Reply:
No, constexpr implies inline.

Reply:
That only applies to static member variables, not namespace-scope variables. Namespace-scope variables like `name_v` are going to get duplicated in every translation unit that includes this code, unless they are given the `inline` treatment.

This was the original reason for adopting `name_of<T>()` and `guid_of<T>()` as functions rather than constexpr template variables - our version of MSVC at the time didn't support inline variables, and the bloat was causing our customers pain.

Reply:
Hmm, I'll double check what's going on here.

Reply:
Indeed: https://github.com/microsoft/xlang/pull/605